warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/ProtobufEdgeDataTypeHandler.java

```diff
@@ -421,141 +169,44 @@ public void funnel(Key from, PrimitiveSink into) {
-            return this.disallowlistValueLookup.get(dataType).contains(fieldValue);
-        } else if (disallowlistValueLookup.containsKey(EDGE_DEFAULT_DATA_TYPE)) {
-            // perhaps there is no disallowlist, which is fine
-            return this.disallowlistValueLookup.get(EDGE_DEFAULT_DATA_TYPE).contains(fieldValue);
-        }
-        return false;
-    }
-
-    // used so we don't write duplicate stats entries for events with multiple field values;
-    protected Set<HashCode> activityLog = null;
-    protected Set<HashCode> durationLog = null;
-
-    protected BloomFilter<Key> activityLogBloom = null;
-    protected BloomFilter<Key> durationLogBloom = null;
-
     @Override
     public long process(KEYIN key, RawRecordContainer event, Multimap<String,NormalizedContentInterface> fields,
                     TaskInputOutputContext<KEYIN,? extends RawRecordContainer,KEYOUT,VALUEOUT> context, ContextWriter<KEYOUT,VALUEOUT> contextWriter)
                     throws IOException, InterruptedException {
-        long edgesCreated = 0;
-        long activityDate = -1;
-        boolean validActivityDate = false;
-        boolean activityEqualsEvent = false;
-        String edgeAttribute2 = null, edgeAttribute3 = null;
-
-        String loadDateStr = null;
 
-        if (event.fatalError()) {
-            return edgesCreated;
-        } // early short circuit return
-
-        // get edge definitions for this event type
+        long edgesCreated = 0;
         String typeName = getTypeName(event.getDataType());
 
-        List<EdgeDefinition> edgeDefs = null;
-        EdgeDefinitionConfigurationHelper edgeDefConfigs = null;
-        if (!edges.containsKey(typeName)) {
-            return edgesCreated; // short circuit, no edges defined for this type
+        // get edge definitions for this event type
+        edgeDefConfigs = edgeConfig.getEdges().get(typeName);
+        List<EdgeDefinition> edgeDefs = edgeDefConfigs.getEdges();
 
+        if (shortCircuit(event, typeName, edgeConfig.getEdges())) {
+            return edgesCreated;
         }
-        edgeDefConfigs = edges.get(typeName);
-        edgeDefs = edgeDefConfigs.getEdges();
 
-        trimFieldGroup = context.getConfiguration().getBoolean(typeName + TRIM_FIELD_GROUP, false);
+        // Track metadata for this event
+        eventMetadataRegistry = new HashMap<>();
 
         /**
          * If enabled, set the filtered context from the NormalizedContentInterface and create the script cache
          */
-        if (evaluatePreconditions) {
+        if (edgeConfig.evaluatePreconditions()) {
             setupPreconditionEvaluation(fields);
-
         }
 
-        // Get the load date of the event from the fields map
-        loadDateStr = getLoadDateString(fields);
-
+        boolean trimGroups = context.getConfiguration().getBoolean(typeName + EdgeIngestConfiguration.TRIM_FIELD_GROUP, false);
+        EdgeEventFieldUtil edgeEventFieldUtil = new EdgeEventFieldUtil(trimGroups);
```

Comment:
Also to avoid object creation and collection, I would consider creating a reusable EdgeEventFieldUtil and clearing it here instead of creating a new one.

Reply:
Perhaps this suggestion is overkill.  Either way, with the refactor it is probably worth observing performance impact.  It would be cool if we had a JMH test handy for the datatype handler that would give us microbenchmarking information for process, given that it's called so much.